doc/whatsnew/fragments/7368.false_positive

```diff
@@ -0,0 +1,3 @@
+Fix ``used-before-assignment`` for functions/classes defined in type checking guard.
```

Comment:
@Pierre-Sassoulas 

These are incorrect and won't and up in the changelog I think